src/storage/transaction/TransactionManager.cpp

```diff
@@ -0,0 +1,493 @@
+}
+
+folly::Future<cpp2::ErrorCode> TransactionManager::resumeTransaction(size_t vIdLen,
+                                                                     GraphSpaceID spaceId,
+                                                                     std::string lockKey,
+                                                                     std::string lockVal,
+                                                                     ResumedResult result) {
+    // 1st, set memory lock
+    auto localKey = NebulaKeyUtils::toEdgeKey(lockKey);
+    CHECK(NebulaKeyUtils::isEdge(vIdLen, localKey));
+    int64_t ver = NebulaKeyUtils::getVersion(vIdLen, localKey);
+    auto keyWoVer = NebulaKeyUtils::keyWithNoVersion(localKey).str();
+    if (FLAGS_enable_multi_versions && !memLock_.insert(std::make_pair(keyWoVer, ver)).second) {
+        return folly::makeFuture(cpp2::ErrorCode::E_MUTATE_EDGE_CONFLICT);
+    }
+
+    // 2nd, get values from remote in-edge
+    auto spPromiseVal = std::make_shared<cpp2::ErrorCode>(cpp2::ErrorCode::SUCCEEDED);
+    auto c = folly::makePromiseContract<cpp2::ErrorCode>();
+    auto remoteKey = reverseKey(vIdLen, spaceId, localKey, ver);
+    interClient_->getValue(vIdLen, spaceId, remoteKey)
+        .via(exec_.get())
+        .thenValue([=, p = std::move(c.first)](auto&& errOrVal) mutable {
+            if (!nebula::ok(errOrVal)) {
+                *spPromiseVal = nebula::error(errOrVal);
+                return;
+            }
+            auto lockedPtr = result->wlock();
+            if (!lockedPtr) {
+                *spPromiseVal = cpp2::ErrorCode::E_MUTATE_EDGE_CONFLICT;
+                return;
+            }
+            auto& kv = *lockedPtr;
+            if (FLAGS_enable_multi_versions) {
+                auto verPrev =
+                    NebulaKeyUtils::isEdge(vIdLen, kv.first)
+                        ? folly::Endian::big(NebulaKeyUtils::getVersion(vIdLen, kv.first))
+                        : std::numeric_limits<int64_t>::max();
+                auto verCurr = folly::Endian::big(ver);
+                LOG_IF(INFO, FLAGS_trace_toss)
+                    << "compare ver=" << ver << ", verCurr=" << verCurr << ", verPrev=" << verPrev;
+                if (verCurr > verPrev) {
+                    *spPromiseVal = cpp2::ErrorCode::E_OUTDATED_LOCK;
+                    return;
+                }
+            }
+            kv.first = localKey;
+            kv.second = nebula::value(errOrVal);
+            // 3rd, commit local key(indexes)
+            auto localPart = NebulaKeyUtils::getPart(localKey);
+            /*
+             * if mvcc enabled, get value will get exactly the same ver in-edge against lock
+             *                  which means we can trust the val of lock as the out-edge
+             *              else, don't trust lock.
+             * */
+            if (FLAGS_enable_multi_versions) {
+                commitBatch(spaceId, localPart, lockVal)
```

Comment:
The `value `part of the lock data is the same as the value part of the normal edge data, right?
But Why not here？

Reply:
talked offline, the data is sth special, not normal data.